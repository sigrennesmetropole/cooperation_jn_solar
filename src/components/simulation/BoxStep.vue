<script lang="ts" setup>
import { UiButton } from '@sigrennesmetropole/cooperation_jn_common_ui'

const props = defineProps<{
  activeButton: boolean
}>()
</script>

<template>
  <div
    class="flex flex-col items-center justify-center border border-slate-600 rounded-lg px-6"
  >
    <slot name="nameOfStep"></slot>
    <slot name="image"></slot>
    <slot name="text"></slot>
    <UiButton
      class="mt-4 mb-6 ui-btn-primary gap-3 px-4 py-3 h-12 w-full"
      :disabled="!props.activeButton"
      @click="$emit('buttonBoxAction')"
    >
      <slot name="buttonContent"></slot>
    </UiButton>
  </div>
</template>
