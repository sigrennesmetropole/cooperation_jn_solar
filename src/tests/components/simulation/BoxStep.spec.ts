import { mount } from '@vue/test-utils'
import BoxStep from '@/components/simulation/BoxStep.vue'

describe('BoxStep', () => {
  it('renders correctly', () => {
    const wrapper = mount(BoxStep, {
      slots: {
        nameOfStep: '<div>Name of Step</div>',
        image: '<img src="test-image.jpg" alt="Test Image">',
        text: '<p>Some text</p>',
        buttonContent: '<span>Click me</span>',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('emits buttonBoxAction event when the button is clicked', async () => {
    const wrapper = mount(BoxStep, {
      slots: {
        nameOfStep: '<div>Name of Step</div>',
        image: '<img src="test-image.jpg" alt="Test Image">',
        text: '<p>Some text</p>',
        buttonContent: '<span>Click me</span>',
      },
      props: {
        activeButton: true,
      },
    })

    const button = wrapper.find('button')
    await button.trigger('click')

    expect(wrapper.emitted('buttonBoxAction')).toBeTruthy()
  })
})
