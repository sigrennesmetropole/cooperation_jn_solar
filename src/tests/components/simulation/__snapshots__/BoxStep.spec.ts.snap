// Vitest Snapshot v1

exports[`BoxStep > renders correctly 1`] = `
"<div class=\\"flex flex-col items-center justify-center border border-slate-600 rounded-lg px-6\\">
  <div>Name of Step</div><img src=\\"test-image.jpg\\" alt=\\"Test Image\\">
  <p>Some text</p><button disabled=\\"\\" class=\\"mt-4 mb-6 ui-btn-primary gap-3 px-4 py-3 h-12 w-full\\">
    <!----><span class=\\"m-auto font-semibold\\"><span>Click me</span></span>
    <!---->
  </button>
</div>"
`;
